src/SelfServe/SqlX/SqlX.rp.ts

```diff
@@ -1,33 +1,78 @@
+  return `/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${name}/services/sqlx`;
+};
+
+export const updateDedicatedGatewayResource = async (sku: string, instances: number): Promise<void> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  const body: UpdateDedicatedGatewayRequestParameters = {
+    properties: {
+      instanceSize: sku,
+      instanceCount: instances,
+      serviceType: "Sqlx",
+    },
+  };
+  return armRequest({ host: configContext.ARM_ENDPOINT, path, method: "PUT", apiVersion, body });
 };
 
-export const getRegionSpecificMaxInstances = async (): Promise<number> => {
-  // TODO: write RP call to get max number of instances needed for this region
-  throw new Error("getRegionSpecificMaxInstances not implemented");
+export const deleteDedicatedGatewayResource = async (): Promise<void> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  return armRequest({ host: configContext.ARM_ENDPOINT, path, method: "DELETE", apiVersion });
 };
 
-export const updateDedicatedGatewayProvisioning = async (sku: string, instances: number): Promise<void> => {
-  // TODO: write RP call to update dedicated gateway provisioning
-  throw new Error(
-    `updateDedicatedGatewayProvisioning not implemented. Parameters- sku: ${sku}, instances:${instances}`
-  );
+export const getDedicatedGatewayResource = async (): Promise<SqlxServiceResource> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  return armRequest<SqlxServiceResource>({ host: configContext.ARM_ENDPOINT, path, method: "GET", apiVersion });
 };
 
-export const initializeDedicatedGatewayProvisioning = async (): Promise<DedicatedGatewayResponse> => {
-  // TODO: write RP call to initialize UI for dedicated gateway provisioning
-  throw new Error("initializeDedicatedGatewayProvisioning not implemented");
+export const getCurrentProvisioningState = async (): Promise<DedicatedGatewayResponse> => {
+  try {
+    const response = await getDedicatedGatewayResource();
+    return {
+      sku: response.properties.instanceSize,
+      instances: response.properties.instanceCount,
+      status: response.properties.status,
+      endpoint: response.properties.sqlxEndPoint,
+    };
+  } catch (e) {
+    return { sku: undefined, instances: undefined, status: undefined, endpoint: undefined };
+  }
 };
 
 export const refreshDedicatedGatewayProvisioning = async (): Promise<RefreshResult> => {
-  // TODO: write RP call to check if dedicated gateway update has gone through
-  throw new Error("refreshDedicatedGatewayProvisioning not implemented");
+  try {
+    const response = await getDedicatedGatewayResource();
+    if (response.properties.status === ResourceStatus.Running.toString()) {
```

Comment:
cleaner if you convert to status to enum and then do a switch on the enum
    const status = ResourceStatus[response.properties.status as keyof typeof ResourceStatus];
    switch(status) {
      case ResourceStatus.Running:
        return {isUpdateInProgress: false, notificationMessage: undefined}
.
.
.
      default:
        return {isUpdateInProgress: true, notificationMessage: "UpdateMessage"}
    }